app/models/user.rb

```diff
@@ -110,6 +112,18 @@ def validate
   before_validation :dummy_password_for_external_auth
   before_destroy :destroy_subscribed_widget_sets
 
+  def check_reference
+    present_ref = []
+    %w(miq_requests vms miq_widgets miq_templates).each do |model|
+      present_ref << model.classify unless public_send(model).first.nil?
+    end
+
+    if present_ref.size.positive?
+      errors.add(:users, "reference to this user with id [#{id}] present in other models: #{present_ref.join(" ")}")
```

Comment:
What does this error look like in the UI? 

I believe it will become `"User: Users reference to this user with id [1] present in other models: miq_requests"

Maybe something like this is better since you're not validating a :users attribute (you can use base)

`.errors.add(:base, "user with id [#{u.id}] has references to other models: #{present_ref.join(' ')}")`

```
irb(main):034:0> u2.errors.add(:base, "user with id [#{u.id}] has references to other models: #{present_ref.join(' ')}")
=> ["user with id [1] has references to other models: MiqRequest"]
irb(main):035:0> u2.errors.full_messages
=> ["user with id [1] has references to other models: MiqRequest"]
```

Reply:
in UI it shows: 
```
EVM User "yuri": Error during delete: reference to this user with id [20] present in other models: MiqRequest
```

Reply:
will correct message, it does look broken